internal/controllers/apis/imagemap/imagemap.go

```diff
@@ -0,0 +1,31 @@
+package imagemap
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/tilt-dev/tilt/pkg/apis/core/v1alpha1"
+	"github.com/tilt-dev/tilt/pkg/model"
+)
+
+// Inject completed ImageMaps into the environment of a local process that
+// wants to deploy to a cluster..
+//
+// Current env vars:
+// TILT_IMAGE_i - The reference to the image #i from the point of view of the cluster container runtime.
+// TILT_IMAGE_MAP_i - The name of the image map #i with the current status of the image.
```

Comment:
Is the intended usage here that you'd know your image map name, so iterate through the `TILT_IMAGE_MAP_i` variables until you find the one you want and then get the value from the corresponding `TILT_IMAGE_i`?

Reply:
Good question! I'm hoping that in most cases, it will look something like this:

```
k8s_custom_deploy(
   image_deps=['be', fe'],
   apply_cmd="""
./inject.sh "$TILT_IMAGE_0"
./inject.sh "$TILT_IMAGE_1"
./deploy.sh
""")
```

or use `tilt get imagemap "$TILT_IMAGE_MAP_0"` to get the full imagemap data

i know Make + Bazel both use the approach of having a variable expansion syntax 
https://docs.bazel.build/versions/main/be/make-variables.html

so i guess we could also do something like `$(image fe)` to expand to the full image name BEFORE the command runs.

But i didn't want to go down that path quite yet...having numbered env variables seemed a good place to start that let us change things later.

Reply:
Ah right, the order is deterministic, so that makes it easy.

Definitely agree that adding another level of variable expansion is excessive for the moment, but that's pretty neat (and would definitely help with POSIX vs Windows 😬)